tools/internal_ci/linux/grpc_xds_url_map_python.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+# Copyright 2021 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex -o igncr || set -ex
```

Comment:
This line was changed from the original build file in java repo, while moving it to core: #26033. @yashykt did that to be consistent with the rest of build files in this repo.

In retrospect, I should've checked the logs more thoroughly:

> `github/grpc/tools/internal_ci/linux/grpc_xds_url_map.sh: line 16: set: igncr: invalid option name`

Let's revert it to

```suggestion
set -eo pipefail
```

- We don't need `igncr`: we don't run this on cygwin
- `-x` is turned on in of in `run_test` on demand
- `pipefail` is a [commonly](https://buildkite.com/docs/pipelines/writing-build-scripts) recommended [safety](https://sipb.mit.edu/doc/safe-shell/) feature that ensures bash propagates (error) exit codes through pipe. People often wrongly assume this is how shell scripts work in the first place. Been there myself.

Updating other files is out of scope of this PR, but let's at least start using this opts in the new build scripts we add.